### PR TITLE
[DOR-179][AO] link errors to the date sub-fields

### PR DESCRIPTION
**Pull request by arturopala**
This PR:
- introduces a more precise error linking for multi-input fields, like date or time,
- reimplements grouping of `.invalid-*` errors to include all of them, not only `.invalid-values`

### CODE REVIEW

File: test/uk/gov/hmrc/traderservices/controllers/Time24FieldHelperSpec.scala [hunk 1]
```diff
@@ -83,33 +83,33 @@ class Time24FieldHelperSpec extends UnitSpec with FormMappingMatchers {
       val validate = t => validTimeFields("foo", required = true)(t)
       validate(("12", "00")) shouldBe Valid
       validate(("12", "00")) shouldBe Valid
-      validate(("", "")) shouldBe Invalid(ValidationError("error.foo.all.required"))
-      validate(("", "")) shouldBe Invalid(Seq(ValidationError("error.foo.all.required")))
+      validate(("", "")) shouldBe Invalid(ValidationError(Seq("subfieldFocus=hour", "error.foo.all.required")))
+      validate(("", "")) shouldBe Invalid(Seq(ValidationError(Seq("subfieldFocus=hour", "error.foo.all.required"))))
```
Comment (Tasmine10ten) [inline test/uk/gov/hmrc/traderservices/controllers/Time24FieldHelperSpec.scala:87]: Some duplicate tests on this page?

File: test/uk/gov/hmrc/traderservices/controllers/VesselDetailsFormSpec.scala [hunk 2]
```diff
@@ -117,7 +118,7 @@ class VesselDetailsFormSpec extends UnitSpec with FormMatchers {
         .updated("timeOfArrival.minutes", "")
       form.bind(input).value shouldBe None
       form.bind(input).errors should haveOnlyErrors(
-        FormError("timeOfArrival", "error.timeOfArrival.all.required")
+        FormError("timeOfArrival", Seq("subfieldFocus=hour", "error.timeOfArrival.all.required"))
```
Comment (Tasmine10ten) [inline test/uk/gov/hmrc/traderservices/controllers/VesselDetailsFormSpec.scala:121]: Time of arrival is fully missing in this test

Comment (arturopala) [reply test/uk/gov/hmrc/traderservices/controllers/VesselDetailsFormSpec.scala:121]: ???

File: test/uk/gov/hmrc/traderservices/controllers/Time24FieldHelperSpec.scala (hunk 1 above)
Comment (arturopala) [reply test/uk/gov/hmrc/traderservices/controllers/Time24FieldHelperSpec.scala]: fixed

**Review by Tasmine10ten — APPROVED** [review]

**Pull request merged**